### PR TITLE
prov/psm2: Enable completion polling optimization when PSM2 source is compiled in

### PR DESCRIPTION
**Pull request by j-xiong**
Include bug fix to the makefile, addition to the configure script, and new / updated completion polling macros.

### CODE REVIEW

File: prov/psm2/configure.m4 [hunk 1]
```diff
@@ -74,3 +75,29 @@ AC_ARG_WITH([psm2-src],
...
+	infile1=$srcdir/prov/psm2/src/psm2/psm_mq_internal.h
+	infile2=$srcdir/prov/psm2/src/psm2/psm_ep.h
+	outfile=$srcdir/prov/psm2/src/psm2/psm2_internal.h
+	$as_echo '#include "opa_queue.h"' >$outfile
+	grep '^#define NUM_HASH_CONFIGS' $infile1 >>$outfile
+	grep '^#define NUM_MQ_SUBLISTS' $infile1 >>$outfile
+	grep '^typedef.*callback_fn_t.*psm2_mq_req_t' $infile1 >>$outfile
+	FI_PSM2_GEN_STRUCT_DEF(mqq, $infile2, $outfile)
+	FI_PSM2_GEN_STRUCT_DEF(psm2_mq_req, $infile1, $outfile)
+])
```
Comment (shefty) [inline prov/psm2/configure.m4]: gads - this seems ugly -- can we just copy-paste the structures to an internal header file or something?

Comment (j-xiong) [reply prov/psm2/configure.m4]: It was originally done that way. However, being internal means that the structure may change and generating the definition from the source is more future proof.

**Comment by j-xiong** [comment]
Update the PR to remove the need of generating temporary header file.

**Comment by j-xiong** [comment]
The travis-ci linux clang build failure was due to some unrelated issue:

The command "sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install rpm libibverbs-dev librdmacm-dev libnl-3-200 libnl-3-dev libnl-route-3-200 libnl-route-3-dev libnuma-dev" failed and exited with 100 during ...

Ignore this specific failure.

**Pull request merged**